idseq_pipeline/commands/postprocess_functions.py

```diff
@@ -318,7 +322,45 @@ def run_stage3(lazy_run=False):
         input_m8,
         os.path.join(RESULT_DIR, ALIGN_VIZ_DIR))
 
-    # combine results
+    # run assembly
+    def make_inputs_for_assembly():
+        # Get taxids to assemble based on criteria from report, currently just species taxids with the most reads
+        execute_command("aws s3 cp --quiet %s/%s %s/" % (SAMPLE_S3_INPUT_PATH, NT_JSON, INPUT_DIR))
+        pipeline_output_json = os.path.join(INPUT_DIR, NT_JSON)
+        with open(pipeline_output_json) as f:
+            pipeline_output = json.load(f)
+        taxon_counts = pipeline_output['pipeline_output']['taxon_counts_attributes']
+        eligible_taxa = [item for item in taxon_counts if item['tax_level'] == 1 and int(item['tax_id']) > 0]
+        # max_count = max([item['count'] for item in eligible_taxa])
+        taxids_to_assemble = [item['tax_id'] for item in eligible_taxa if item['count'] >= 100]
+        # Get reads for those taxids
+        output = {}
+        full_fasta = os.path.join(RESULT_DIR, TAXID_ANNOT_SORTED_FASTA_NT)
+        delimiter = 'species_nt'
+        for taxid in taxids_to_assemble:
+            partial_fasta = taxid + ".fasta"
+            subprocess.check_call("grep -A 1 --no-group-separator '%s:%s:' %s > %s" % (delimiter, taxid, full_fasta, partial_fasta), shell=True)
+            output[taxid] = partial_fasta
+        return output        
+
+    def spades(input_fasta, output_fasta):
+        tmp_output_dir = input_fasta + "_temp_output"
+        try:
+            execute_command_realtime_stdout("spades.py -s %s -o %s -m 60 -t 32 --only-assembler" % (input_fasta, tmp_output_dir))
+            subprocess.check_call("mv %s/scaffolds.fasta %s" % (tmp_output_dir, output_fasta), shell=True)
+            return True
+        except:
+            return False
+
+    inputs = make_inputs_for_assembly()
+    for taxid, input_fasta in inputs.iteritems():
+        output_fasta = os.path.join(RESULT_DIR, taxid + ".scaffolds.fasta")
+        if spades(input_fasta, output_fasta):
```

Comment:
If 'spades.py' or 'mv %s/scaffolds.fasta' fail and return False, will that still be logged properly?